sources/packages/backend/libs/integrations/src/sfas-integration/sims-sfas-files/sims-to-sfas-header.ts

```diff
@@ -0,0 +1,38 @@
+import { SpecializedStringBuilder } from "@sims/utilities";
+import { SIMSToSFASBaseRecord } from "./sims-to-sfas-base.record";
+
+/**
+ * Header record for SIMS to SFAS file.
+ */
+export class SIMSToSFASHeader extends SIMSToSFASBaseRecord {
+  /**
+   * Specifies who sent this file.
+   */
+  originator: string;
+  /**
+   * Description of this file.
+   */
+  title: string;
+  /**
+   * Date & time that this file was generated in UTC.
+   */
+  creationDate: Date;
+  /**
+   * Get the information as a fixed line format to be
+   * added to the file uploaded to the SFTP.
+   * @returns fixed line formatted.
+   */
+  getFixedFormat(): string {
+    const record = new SpecializedStringBuilder({
+      stringFiller: this.spaceFiller,
+      numberFiller: this.numberFiller,
+      dateFiller: this.spaceFiller,
+      dateFormat: this.dateFormat,
+    });
```

Comment:
As a suggestion, the base class can have a method called `createQueryBuilder` that will return a newly configured query builder, avoiding having `spaceFiller`, `numberFiller`, and `dateFormat` as protected properties.

Reply:
I saw the change but I would still see it as a method to create the new string builder instead of a read-only property.

Reply:
Agreed. It is updated further.

Reply:
I am not sure if we are on the same page, my suggestion was the below.
```ts
protected getStringBuilder(): SpecializedStringBuilder {
    const dateFormat = "YYYYMMDD";
    const spaceFiller = " ";
    const numberFiller = "0";
    return new SpecializedStringBuilder({
      stringFiller: spaceFiller,
      numberFiller: numberFiller,
      dateFiller: spaceFiller,
      dateFormat: dateFormat,
    });
  }
```
Then in the consumer part.
```ts
const record = this.getStringBuilder();
```

Reply:
That is exactly what I did. 
![image](https://github.com/user-attachments/assets/e7a92385-61a2-4bc7-830c-f066ecc4e18c)

Reply:
Still, not what I meant. Lets chat.

Reply:
Got it after syncup. Thanks.